t/cmd/git-credential-lfstest.go

```diff
@@ -51,8 +51,10 @@ func (c *credential) Serialize(capabilities map[string]struct{}, state []string)
 				}
 			}
 		}
-	} else if len(c.username) != 0 && len(c.password) != 0 {
-		creds["username"] = []string{c.username}
+	} else if len(c.authtype) == 0 && (len(username) == 0 || username[0] == c.username) {
+		if len(username) == 0 {
+			creds["username"] = []string{c.username}
+		}
```

Comment:
TBH the only thing that I am still puzzled about is why we check for `len(c.authtype) == 0` here.